js/modules/k6/ws/ws.go

```diff
@@ -155,17 +155,11 @@ func (mi *WS) Connect(url string, args ...goja.Value) (*WSHTTPResponse, error) {
 					header.Set(key, headersObj.Get(key).String())
 				}
 			case "tags":
-				tagsV := params.Get(k)
-				if goja.IsUndefined(tagsV) || goja.IsNull(tagsV) {
-					continue
-				}
-				tagObj := tagsV.ToObject(rt)
-				if tagObj == nil {
-					continue
-				}
-				for _, key := range tagObj.Keys() {
-					tags = tags.With(key, tagObj.Get(key).String())
+				newTags, err := common.ApplyCustomUserTags(rt, tags, params.Get(k))
+				if err != nil {
+					return nil, fmt.Errorf("invalid ws.Connect() metric tags: %w", err)
```

Comment:
```suggestion
					return nil, fmt.Errorf("invalid ws.connect() metric tags: %w", err)
```

https://k6.io/docs/javascript-api/k6-ws/connect/